sites/docs/src/content/docs/tutorials/tests_and_test_data/components/03_testing_modules.md

```diff
@@ -0,0 +1,810 @@
+            """
+            input[0] = Channel.fromList(...)
+            """
+            }
+        }
+    }
+
+    test("Test with three different inputs") {
+        when {
+            process {
+                """
+                input[0] = UNTAR1.out.untar.map{ it[1] }
+                input[1] = UNTAR2.out.untar.map{ it[1] }
+                input[2] = UNTAR3.out.untar.map{ it[1] }
+                """
+            }
+        }
+
+        then {
+            // ...
+        }
+    }
+}
+```
+
+### Testing parameter variations
+
+Some modules may require additional parameters added to the test command to successfully run.
+
+These can be specified using a `params` input and an `ext.args` variable within the process scope of the `nextflow.config` file, which exists alongside the test files themselves (and is automatically loaded when the test workflow `main.nf` is executed).
+
+If your module requires a `nextflow.config` file to run, create the file to the module’s `tests/` directory and add the following code to use parameters defined in the `when` scope of the test.
+
+```bash
+touch modules/nf-core/<tool>/<subtool>/tests/nextflow.config
+```
+
+```nextflow.config
+process {
+  withName: 'MODULE' {
+    ext.args = params.module_args
+  }
+}
+```
+
+You do not need to modify the contents of this file any further.
+
+Then add the config to the `main.nf.test` file and supply the params in the `when` section of the test.
+
+```main.nf.test
+process "MODULE"
+config "./nextflow.config"
+
+when {
+  params {
+    module_args = '--extra_opt1 --extra_opt2'
+  }
+  process {
+    """
+    input[0] = [
+      [ id:'test1', single_end:false ], // meta map
+      file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
+    ]
+    """
+  }
+}
+```
+
+### Choosing Parameter Configuration Methods
+
+**Use `nextflow.config` when:**
+
+- Multiple tests in the same module need the same parameter structure
+- You need complex process-specific configurations (memory, cpus, etc.)
+- Parameters require process selectors or conditional logic
+- You want to maintain consistent configuration across all tests
+
+**Use inline `params` blocks when:**
+
+- Testing different parameter values in individual tests
+- You need test-specific parameter overrides
+- Parameters are simple and don't require process configuration
+- You want to keep test parameters close to the test logic
+
+**Example comparison:**
+
+```groovy
+// nextflow.config approach - good for consistent configuration
+process {
+  withName: 'BLAST_BLASTN' {
+    ext.args = params.blast_args
+    memory = params.blast_memory ?: '8.GB'
+  }
+}
+
+// Inline params approach - good for test-specific values
+test("custom evalue") {
+    when {
+        params {
+            blast_args = '-evalue 0.001 -max_target_seqs 10'
+        }
+        process {
+            // test implementation
+        }
+    }
+}
+```
+
```

Comment:
```suggestion
```

O think you can delete this comparison as you don't introduce the config only method?